contracts/drafts/Strings.sol

```diff
@@ -0,0 +1,39 @@
+pragma solidity ^0.5.0;
+
+/**
+ * @title Strings
+ * @dev String operations.
+ */
+library Strings {
+    /**
+     * @dev Concatenates two strings.
+     */
+    function concatenate(string memory a, string memory b) internal pure returns (string memory concatenatedString) {
+        return string(abi.encodePacked(a, b));
+    }
+
+    /**
+     * @dev Converts a uint256 to a string.
+     * via OraclizeAPI - MIT licence
+     * https://github.com/oraclize/ethereum-api/blob/b42146b063c7d6ee1358846c198246239e9360e8/oraclizeAPI_0.4.25.sol
+     */
+    function uint256ToString(uint256 value) internal pure returns (string memory) {
+        if (value == 0) {
+            return "0";
+        }
+        uint256 temp = value;
+        uint256 digits;
+        while (temp != 0) {
+            digits++;
+            temp /= 10;
+        }
+        bytes memory buffer = new bytes(digits);
```

Comment:
To avoid the two separate loops we can use a fixed size buffer, because the number is bounded by `uint256` range. It comes out to at most 78 decimal digits.

Then we can track the number of digits in the second while loop to return the appropriate substring.

Reply:
I have removed concatenate from Strings.sol and just left a comment and left concatenate in StringsMock.sol

I wasn't sure how to do the copy so have left the uint256ToString as is for the moment.